### DISCUSSION TITLE
zsh bug

### DISCUSSION THREAD

**Discussion by rwmitchell** [ohmyzsh/ohmyzsh#12673] [Help] [answered]
This is basic ZSH question, not OMZ.
Does this work as anticipated?

`[[ true ]] && { printf "True\n"; false } || { printf "FALSE\n" }`

I get:

> True
FALSE

Is this the correct result?

**Comment by carlosala** [comment, marked as answer]
Indeed, it is expected. AND has higher precedence than OR (like multiplication with sum). And, therefore, first gets executed `[[ true ]]`, then the first print (whose block returns `false`) and then the last block, executed because the first block returns falsy.

**Reply by rwmitchell** [reply]
got it, the braces mean nothing.

**Reply by mcornella** [reply]
POSIX behavior btw, bash does the same: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_09_03. For the desired effect, I guess spliting this up or using if else is necessary.
